lib/src/components/select_input/select_input.dart

```diff
@@ -0,0 +1,648 @@
+import 'package:flutter/foundation.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+
+import '../../../zeta_flutter.dart';
+
+enum _MenuPosition { top, bottom }
+
+/// Class for [ZetaSelectInput]
+class ZetaSelectInput extends StatefulWidget {
+  ///Constructor of [ZetaSelectInput]
+  const ZetaSelectInput({
+    super.key,
+    required this.items,
+    this.onChanged,
+    this.onTextChanged,
+    this.selectedItem,
+    this.size,
+    this.leadingIcon,
+    this.label,
+    this.hint,
+    this.enabled = true,
+    this.rounded = true,
+    this.hasError = false,
+    this.errorText,
+  });
+
+  /// Input items as list of [ZetaSelectInputItem]
+  final List<ZetaSelectInputItem> items;
+
+  /// Currently selected item
+  final ZetaSelectInputItem? selectedItem;
+
+  /// Handles changes of select menu
+  final ValueSetter<ZetaSelectInputItem?>? onChanged;
+
+  /// Handles changes of input text
+  final ValueSetter<String>? onTextChanged;
+
+  /// Determines the size of the input field.
+  /// Default is `ZetaDateInputSize.large`
+  final ZetaWidgetSize? size;
+
+  /// The input's leading icon.
+  final Widget? leadingIcon;
+
+  /// If provided, displays a label above the input field.
+  final Widget? label;
+
+  /// If provided, displays a hint below the input field.
+  final String? hint;
+
+  /// Determines if the input field should be enabled (default) or disabled.
+  final bool enabled;
+
+  /// Determines if the input field should be displayed in error style.
+  /// Default is `false`.
+  /// If `enabled` is `false`, this has no effect.
+  final bool hasError;
+
+  /// In combination with `hasError: true`, provides the error message
+  /// to be displayed below the input field.
+  final String? errorText;
+
+  /// {@macro zeta-component-rounded}
+  final bool rounded;
+
+  @override
+  State<ZetaSelectInput> createState() => _ZetaSelectInputState();
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties
+      ..add(DiagnosticsProperty<bool>('rounded', rounded))
+      ..add(
+        ObjectFlagProperty<ValueSetter<ZetaSelectInputItem?>?>.has(
+          'onChanged',
+          onChanged,
+        ),
+      )
+      ..add(EnumProperty<ZetaWidgetSize?>('size', size))
+      ..add(StringProperty('hint', hint))
+      ..add(DiagnosticsProperty<bool>('enabled', enabled))
+      ..add(DiagnosticsProperty<bool>('hasError', hasError))
+      ..add(StringProperty('errorText', errorText))
+      ..add(ObjectFlagProperty<ValueSetter<String>?>.has('onTextChanged', onTextChanged));
+  }
+}
+
+class _ZetaSelectInputState extends State<ZetaSelectInput> {
+  final OverlayPortalController _overlayController = OverlayPortalController();
+  final _link = LayerLink();
+  late String? _selectedValue;
+  late List<ZetaSelectInputItem> _menuItems;
+  Size _menuSize = const Size(240, 240);
```

Comment:
why is 240 the initial size here?

Reply:
This is only to initialise with some non-null value, and it never takes effect, because the actual `Size` is calculated inside `onToggleMenu` before the menu is shown.

Reply:
It could also be `Size.zero`